### DISCUSSION TITLE
Indication for support of MySQL/MariaDB

### DISCUSSION THREAD

**Discussion by tarilabs** [kubeflow/pipelines#11224] [General]
Hi 👋 
I'm opening this thread for a question we have from the Model Registry WG.

Per the [MR proposal](https://github.com/kubeflow/community/pull/682) we are wrapping MLMD; in the Manifests, we re-use same images defined by KFP Manifests, namely:
- MySQL `mysql:8.0.3`
example: https://github.com/kubeflow/manifests/blob/35539f162ea7fafc8c5035d8df0d8d8cf5a9d327/apps/pipeline/upstream/base/metadata/overlays/db/metadata-db-deployment.yaml#L24
- MLMD `gcr.io/tfx-oss-public/ml_metadata_store_server:1.14.0`
example: https://github.com/kubeflow/manifests/blob/35539f162ea7fafc8c5035d8df0d8d8cf5a9d327/apps/pipeline/upstream/base/metadata/base/metadata-grpc-deployment.yaml#L25

We're interested to know which versions for MySQL, and/or MariaDB, are supported and compatible with MLMD.

We're wondering how to best update the MySQL image, since that is being reported by the Platform WG as being subject of a [number of CVEs](https://blog.kubeflow.org/kubeflow-1.9-release/#cve-scanning).

Given it's an image that is _also_ shared with this WG, it was worth asking in this thread :) 

ps.: I should also note that I acknowledge in the Kubeflow installation example, the MySQL image used is actually `gcr.io/ml-pipeline/mysql:8.0.26` but the question remains.

Thank you in advance for your feedback!

**Comment by gregsheremeta** [comment]
@chensun is there someone at Google that you could ask about this, or might you know?

**Comment by juliusvonkohout** [comment]
> We're wondering how to best update the MySQL image, since that is being reported by the Platform WG as being subject of a [number of CVEs](https://blog.kubeflow.org/kubeflow-1.9-release/#cve-scanning).

Yes please :-)

**Comment by tarilabs** [comment]
@kubeflow/wg-pipeline-leads any news, please? thanks in advance 🙏

**Comment by dandawg** [comment]
The `mysql:8.0.3`  image is used only in `manifests/kustomize/base/metadata/overlays/db/metadata-db-deployment.yaml `, which is only used in the the `platform-agnostic-multi-user-legacy`  deployment. As I understand, this deployment is for supporting the docker executor in the old Argo Workflows deployment (See [here](https://www.kubeflow.org/docs/components/pipelines/legacy-v1/installation/choose-executor/#docker-executor) for background info), which existed pre-KFP-1.8, and pre-Kubernetes 1.20. So it's old, and probably should be fully deprecated at some point.
The `gcr.io/ml-pipeline/mysql:8.0.26` image is used in the current (non-legacy) platform-agnostic deployments. The `mysql:8.0.26` version is still fairly old, but better in my view if only for the fact that it has an arm build that works on Mac chips (whereas `8.0.3`  does not). 

KFP should probably also consider upgrading to `mysql:8.3.x`  or `mysql:8.4.x`  since the `mysql:8.0.x`  versions are already quite old. As far as a specific recommended version of mysql for MLMD, I don't know that.

**Comment by juliusvonkohout** [comment]
@dandawg do you mind just creating a PR to check how far we can upgrade mysql and remove deprecated -legacy stuff?
